src/main/plugins/org.dita.base/xsl/preprocess/topicpullImpl.xsl

```diff
@@ -714,37 +714,55 @@ mode="topicpull:figure-linktext" and mode="topicpull:table-linktext"
                          contains(@class,' topic/indextermref ')]"
                 mode="topicpull:get-stuff_get-shortdesc" priority="10"/>
 
-  <!-- Get the link text for a link or cross reference. If specified locally, use that. Otherwise,
-       work with the target to get the text. -->
+  <!-- Get the link text for a cross reference (not a link element). If specified locally, use that.
+       Otherwise, work with the target to get the text. -->
   <xsl:template match="*" mode="topicpull:get-stuff_get-linktext">
     <xsl:param name="targetElement" as="element()"/>
     
     <xsl:choose>
-      <xsl:when test="contains(@class,' topic/link ') and *[not(contains(@class, ' topic/desc '))]">
-        <xsl:apply-templates select="." mode="topicpull:add-usertext-PI"/>
-        <xsl:apply-templates select="*[not(contains(@class, ' topic/desc '))]|comment()|processing-instruction()"/>
-      </xsl:when>
-      <xsl:when test="dita-ot:is-link(.) and (normalize-space(.) != '' or *[not(contains(@class, ' topic/desc '))])">
+      <xsl:when test="dita-ot:is-link(.) and (normalize-space(text()) != '' or *[not(contains(@class, ' topic/desc ') or contains(@class,' ditaot-d/ditaval-startprop ') or contains(@class,' ditaot-d/ditaval-endprop '))])">
```

Comment:
This line won't work, needs to be fixed.

I switched from `normalize-space(.)` to `normalize-space(text())` because using the current node will give the wrong result when there is a `<desc>` tag -- but `text()` fails when there are sub-elements and you get more than one text node.

Reply:
Has this been fixed?

Reply:
Now fixed, and split into multiple lines.